File: 02_recursion_basics/02_06_reverse_array/ReverseArray.java
public class ReverseArray {
    public static void main(String[] args) {
        int arr[] = {1, 2, 3, 4, 5, 6};

        System.out.println("Before:-");
        for (int i : arr) {
            System.out.print(i + " ");
        }
        
        reverseArray(0, arr.length - 1, arr);
        System.out.println();

        System.out.println("After:-");
        for (int i : arr) {
            System.out.print(i + " ");
        }
    }

    private static void reverseArray(int start, int end, int arr[]) {
        if (start < end) {
            int temp = arr[end];
            arr[end] = arr[start];
            arr[start] = temp;
            reverseArray(start + 1, end - 1, arr);
        }
    }
}

File: 02_recursion_basics/02_06_reverse_array/reverse_array.cpp
#include<iostream>

using namespace std;

void reverseArray(int start, int end, int arr[]) {
    if (start < end) {
        swap(arr[start], arr[end]);
        reverseArray(start + 1, end - 1, arr);
    }
}

int main() {
    int n = 5;
    int arr[] = {1, 2, 3, 4, 5};

    cout << "Before:-" << endl;
    for (size_t i = 0; i < n; i++)
    {
        cout << arr[i] << " ";
    }
    
    reverseArray(0, n - 1, arr);
    
    cout << endl;
    cout << "After:-" << endl;
    for (size_t i = 0; i < n; i++)
    {
        cout << arr[i] << " ";
    }
    
}